src/components/Select/ComboboxMulti/ComboboxMulti.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { IoIosArrowUp, IoIosArrowDown } from 'react-icons/io';
+import bem from '../../../utils/bem/bem';
+import { Props as SelectBaseProps, SelectBase } from '../SelectBase';
+import { SuggestionsList } from '../SuggestionsList';
+import styles from './ComboboxMulti.scss';
+import { ESCAPE_KEY, TAB_KEY } from '../../../constants';
+
+interface Props<S>
+    extends Omit<
+        Omit<Omit<SelectBaseProps<S>, 'listRenderer'>, 'focusedRenderer'>,
+        'blurredRenderer'
+    > {
+    inputPlaceholder: string;
+}
+
+const { elem } = bem('ComboboxMulti', styles);
+
+export function ComboboxMulti<S>(props: Props<S>) {
+    const {
+        onSelectionChange,
+        inputRef: inputRefFromProps,
+        suggestions,
+        suggestionToString,
+        showClearButton,
+        noSuggestionsPlaceholder,
+        onBlur,
+        onInputValueChange,
+        onClearAllSelected,
+        inputPlaceholder,
+        ...rest
+    } = props;
+    const inputRef = inputRefFromProps || React.createRef<HTMLInputElement>();
+
+    // eslint-disable-next-line react/display-name
+    const renderFocused = ({ getInputProps, getToggleButtonProps, onBlur: blur }) => {
+        const handleInputKeyDown = event => {
+            if (event.key === TAB_KEY) {
+                inputRef.current?.blur();
+                blur();
+            } else if (event.key === ESCAPE_KEY) {
+                // prevents key propagation and sets the focus on parent component
+                inputRef.current?.blur();
+                blur();
+                inputRef.current?.parentElement?.focus();
+                event.stopPropagation();
+            }
+        };
+
+        return (
+            <div tabIndex={0} role="searchbox" {...elem('wrapper', { ...props })}>
+                <input
+                    {...getInputProps({
+                        ref: inputRef,
+                        placeholder: inputPlaceholder,
+                        onKeyDown: handleInputKeyDown,
+                        ...elem('input', { ...props }),
+                    })}
+                />
+                <div {...elem('buttons', { ...props })}>
+                    <IoIosArrowUp {...getToggleButtonProps()} />
+                </div>
+            </div>
+        );
+    };
+
+    // eslint-disable-next-line react/display-name
+    const renderBlurred = ({ getInputProps, getToggleButtonProps }) => (
```

Comment:
These functions look almost the same. I wonder if it couldn't be simplified into 1 function with some params (mainly for which arrow is displayed) instead. But maybe that would mess up the keyboard things..

Reply:
@eszthoff I'm not sure if that refactoring would be a good thing there. Since those are two separate render functions maybe it would be more convenient to keep them as they are in terms of readability?